### PR TITLE
Add support for OCaml 4.06 and Lwt 3

### PR DESCRIPTION
**Pull request by talex5**
- Adds an `opam` file.
- Uses `lwt_react` instead of `lwt.react`, to support Lwt 3.
- Uses `Bytes` for safe-string support.
- Adds Travis tests.

Note that the Travis tests do not actually run the tests, which fail for me with:

```
=[ garbage collection ]=========================================================
safety check: success
testing garbage collection of a signal without a switch: failure
testing garbage collection of a signal with a switch: success

Test failed.

1 of 5 tests failed.
```

Also, the tests sometimes hang at:

```
=[ communication ]==============================================================
sending and receiving 100 messages through the message bus.
```

I don't think I changed anything related to this. I tested the original sources using 4.02.3 and they always hang for me too. I tried removing the communication test to try the GC tests, but that then failed with:

```
=[ garbage collection ]=========================================================
main.native: obus(transport): autolaunch failed: End_of_file
main.native: obus(bus): Failed to open a connection to the session bus: End_of_file
test failed with: End_of_file
```

So maybe the unit-tests just don't work?

### CODE REVIEW

File: opam [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+authors: ["Jérémie Dimino"]
+homepage: "https://github.com/diml/obus"
+bug-reports: "https://github.com/diml/obus/issues"
+build: [
+  ["ocaml" "setup.ml" "-configure" "--prefix" prefix]
+  ["ocaml" "setup.ml" "-build"]
+]
+remove: [["ocamlfind" "remove" "obus"]]
+depends: [
+  "ocamlfind"
```
Comment (jpdeplaix) [inline opam]: Could you add the ```build``` attribute ?

**Comment by jpdeplaix** [comment]
I see many uses of ```Bytes.unsafe_to_string```. Is there issues in using ```Bytes.to_string``` instead ?

File: opam (hunk 1 above)
Comment (talex5) [reply opam]: Done.

**Comment by talex5** [comment]
`Bytes.to_string` would make an unnecessary copy. It should be clear in each case that the buffer isn't being reused for anything. In any case, this doesn't change the existing behaviour. Maybe one day OCaml will have affine types and we can check this...

**Comment by talex5** [comment]
I've merged it as-is, to keep the previous behaviour but just update for 4.06. If you still want to change the speed/code-clarity trade-off by copying bytes to strings, I think that would be better as a separate PR.

**Pull request merged**